File: solutions/typescript/scrabble-score/1/scrabble-score.ts
function scoreLetter(letter: string): number {
    switch(letter.toUpperCase()){
      case "A":
      case "E":
      case "I":
      case "O":
      case "U":
      case "L":
      case "N":
      case "R":
      case "S":
      case "T":
        return 1;
      case "D":
      case "G":
        return 2;
      case "B":
      case "C":
      case "M":
      case "P":
        return 3;
      case "F":
      case "H":
      case "V":
      case "W":
      case "Y":
        return 4;
      case "K":
        return 5;
      case "J":
      case "X":
        return 8;
      case "Q":
      case "Z":
        return 10;
      default:
        throw new Error("Invalid Letter");
    }
}

export function score(word: string): number {
  if (!word) return 0;
  
  return word.split("").map(c => scoreLetter(c)).reduce((acc, cur) => acc + cur, 0);
}

File: solutions/python/scrabble-score/1/scrabble_score.py
LETTER_SCORES = {
    **dict.fromkeys("AEIOULNRST", 1),
    **dict.fromkeys("DG", 2),
    **dict.fromkeys("BCMP", 3),
    **dict.fromkeys("FHVWY", 4),
    "K": 5,
    **dict.fromkeys("JX", 8),
    **dict.fromkeys("QZ", 10),
}
        

def score(word):
    try:
        return sum(LETTER_SCORES[letter.upper()] for letter in word)
    except KeyError:
        raise ValueError("Invalid Scrabble Character in word")